wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/SwerveFollowerCommand.java

```diff
@@ -0,0 +1,155 @@
+   * @param outputModuleStates                The raw output module states from the
+   *                                          position controllers.
+   * @param requirements                      The subsystems to require.
+   */
+
+  public SwerveFollowerCommand(Trajectory trajectory,
+                               Supplier<Pose2d> pose,
+                               SwerveDriveKinematics kinematics,
+                               PIDController xdController,
+                               PIDController ydController,
+                               ProfiledPIDController thetaController,
+
+                               Consumer<SwerveModuleState[]> outputModuleStates,
+                               Subsystem... requirements) {
+    m_trajectory = requireNonNullParam(trajectory, "trajectory", "SwerveFollowerCommand");
+    m_pose = requireNonNullParam(pose, "pose", "SwerveFollowerCommand");
+    m_kinematics = requireNonNullParam(kinematics, "kinematics", "SwerveFollowerCommand");
+
+    m_xdController = requireNonNullParam(xdController,
+      "xdController", "SwerveFollowerCommand");
+    m_ydController = requireNonNullParam(ydController,
+      "xdController", "SwerveFollowerCommand");
+    m_thetaController = requireNonNullParam(thetaController,
+      "thetaController", "SwerveFollowerCommand");
+
+    m_outputModuleStates = requireNonNullParam(outputModuleStates,
+      "frontLeftOutput", "SwerveFollowerCommand");
+    addRequirements(requirements);
+  }
+
+  @Override
+  public void initialize() {
+    // Sample final pose to get robot rotation
+    m_finalPose = m_trajectory.sample(m_trajectory.getTotalTimeSeconds()).poseMeters;
+
+    m_timer.reset();
+    m_timer.start();
+  }
+
+  @Override
+  public void execute() {
+    double curTime = m_timer.get();
+
+    var desiredState = m_trajectory.sample(curTime);
+    var desiredPose = desiredState.poseMeters;
+
+    var poseError = desiredPose.relativeTo(m_pose.get());
+
+    double targetXVel = m_xdController.calculate(
+        m_pose.get().getTranslation().getX(),
+        desiredPose.getTranslation().getX());
+
+    double targetYVel = m_ydController.calculate(
+        m_pose.get().getTranslation().getY(),
+        desiredPose.getTranslation().getY());
+
+    // The robot will go to the desired rotation of the final pose in the trajectory,
+    // not following the poses at individual states.
+    double targetAngularVel = m_thetaController.calculate(
+        m_pose.get().getRotation().getRadians(),
+        m_finalPose.getRotation().getRadians());
+
+    double dvRef = desiredState.velocityMetersPerSecond;
+
+    targetXVel += dvRef * poseError.getRotation().getCos();
+    targetYVel += dvRef * poseError.getRotation().getSin();
+
+    var targetChassisSpeeds = new ChassisSpeeds(targetXVel, targetYVel, targetAngularVel);
+
+    var targetModuleStates = m_kinematics.toSwerveModuleStates(targetChassisSpeeds);
+
+    m_outputModuleStates.accept(targetModuleStates);
```

Comment:
Why are you not normalizing the module states here?

Reply:
"The static method normalizeWheelSpeeds(SwerveModuleState[], double) from the type SwerveDriveKinematics should be accessed in a static wayJava(603979893)"
We have to normalize in the robot file.

Reply:
Normalize is a static method so we can make a non-static call to it.